components/service/glean/src/main/java/mozilla/components/service/glean/private/LabeledMetricType.kt

```diff
@@ -38,9 +38,9 @@ data class LabeledMetricType<T>(
         private const val MAX_LABELS = 16
         private const val OTHER_LABEL = "__other__"
 
-        // This regex is used for matching against labels and should allow for dots and/or
-        // snake_case.
-        private val labelRegex = Regex("^[a-z_][a-z0-9_]{0,29}(\\.[a-z0-9_]{0,29})*$")
+        // This regex is used for matching against labels and should allow for dots,
+        // snake_case and/or hyphens.
```

Comment:
One minor nit here, it's calling out allowing `_` by using the term "snake_case", and allowing `-` by saying "hyphens".  It also doesn't call out that the starting character is restricted to a letter or an underscore.  It might be a little more clear if we reworded this to something like:

```
// This regex is used for matching against labels and should allow for dots, underscores,
// and/or hyphens.  Labels are also limited to starting with either a letter or an
// underscore character.
```